### ISSUE TITLE
Report when there's upstream configuration drift

### ISSUE THREAD

**Issue by jsoref** [argoproj/argo-cd#4799]
# Summary

Provide a way to recognize when upstream values' changes aren't being realized.

# Motivation

I had a helm chart which had a values file and an override values file. The override file overrode a value in the main values file. I made a change to the main values file and nothing happened. It took a while of scratching my head to remember the override.

Conceptually, this is the equivalent of missing a security update because one pinned a version.

# Proposal

1. tools like `helm` will need to be improved to offer an instrumentation flag to report a list of overridden values.
1. argo when it runs a command like `helm` can retain the list of overridden values.
1. when argo processes a new run, it should diff it against the previous run, and if there's drift in the overridden values, it could report them.

**Comment by jessesuen** [comment]
> 1. tools like helm will need to be improved to offer an instrumentation flag to report a list of overridden values.

So we can't do anything until the underlying tool supports it?

**Comment by jsoref** [comment]
Yeah, I expect to have to write issues asking for such a feature from the underlying tools... I'll tie them together...
